solr/core/src/java/org/apache/solr/cloud/overseer/ZkStateWriter.java

```diff
@@ -139,8 +139,12 @@ public ClusterState enqueueUpdate(
       throw new IllegalStateException(
           "ZkStateWriter has seen a tragic error, this instance can no longer be used");
     }
-    if (cmds.isEmpty()) return prevState;
-    if (isNoOps(cmds)) return prevState;
+    if (cmds.isEmpty() || isNoOps(cmds)) { // empty and no-ops are valid, should call callback
+      if (callback != null) {
```

Comment:
Thanks @patsonluk. is it possible to add unit test for this method `enqueueUpdate` ?

Reply:
Added `ZkStateWriterTest#testEnqueueCallback`